### ISSUE TITLE
Expression trees support for tuples.

### ISSUE THREAD

**Issue by VSadov** [dotnet/roslyn#12897]
We should consider supporting expressions with tuples in expression tree lambdas.

It can be done either by 
1) Introducing new nodes that represent tuple creation/conversion/field accesses   or
2) Use lowered form

Option 1 is more desirable since it is more expressive, but will require introducing new nodes, possibly as a reducible layer on top of the standard ET nodes.

Option 2 could run into problems where tuple conversions require temporaries and in the ET version (1.0) those are not available. 
It may be possible to use member bindings to avoid this problem, but it is not completely clear if that is possible and it will require that tuple conversion lowering in a context of expression trees to be significantly different.

Option 1 seems more attractive to me at the moment.

**Comment by a-a-k** [comment]
which status on this?

**Comment by evilguest** [comment]
What is preventing this from being implemented via Option 1?

**Comment by VBAndCs** [comment]
Any news about this 4 years old proposal?

**Comment by CyrusNajmabadi** [comment]
> Any news about this 4 years old proposal?

Nope.  No news.

**Comment by sjd2021** [comment]
Is this being worked on?

**Comment by CyrusNajmabadi** [comment]
No.  It is not being worked on.  See https://github.com/dotnet/roslyn/issues/12897#issuecomment-690852209

**Comment by jnm2** [comment]
@CyrusNajmabadi Are you sure? https://github.com/dotnet/csharplang/discussions/4727

**Comment by qsdfplkj** [comment]
This seems to be working:

Expression<Func<int, int, object>> f = (x, y) => Tuple.Create(x, y);

**Comment by poke** [comment]
@qsdfplkj Those are standard method invocations though. This issue proposes tuples on a language level within expression trees.

**Comment by qsdfplkj** [comment]
So using the valuetuple is a good alternative.

**Comment by mguinness** [comment]
Assuming this wasn't considered for C# 14, any likelihood it will be moved out of the backlog for a subsequent version?  It has the most upvotes out of all the [open issues](https://github.com/dotnet/roslyn/issues?q=is%3Aissue%20state%3Aopen%20sort%3Areactions-%2B1-desc).
